### ISSUE TITLE
Potential infinite loop

### ISSUE THREAD

**Issue by airsakarya** [FreeCodeCamp/FreeCodeCamp#5458]
Challenge [Bonfire: Spinal Tap Case](http://www.freecodecamp.com/challenges/bonfire-spinal-tap-case#?solution=function%20spinalCase(str)%20%7B%0A%20%20%0A%20%20for(var%20i%20%3D%200%3B%20i%20%3C%20str.length%3B%20i%2B%2B)%0A%20%20%7B%0A%20%20%20%20var%20char%20%3D%20str.substr(i%2C%201)%3B%0A%20%20%20%20%20if(char%20%3D%3D%3D%20char.toUpperCase()%20%26%26%20char%20!%3D%20%22%20%22)%0A%20%20%20%20%20%7B%0A%20%20%20%20%20%20%20str%20%3D%20str.replaceAt(i%2C%20%22-%22%20%2B%20char)%3B%0A%20%20%20%20%20%7D%0A%20%20%7D%0A%20%20%0A%20%20str%20%3D%20str.toLowerCase().split(%22%20%22).join(%22-%22)%3B%0A%20%20str%20%3D%20str.toLowerCase().split(%22_%22).join(%22-%22)%3B%0A%20%20return%20str%3B%0A%7D%0A%0AString.prototype.replaceAt%3Dfunction(index%2C%20character)%20%7B%0A%20%20%20%20return%20this.substr(0%2C%20index)%20%2B%20character%20%2B%20this.substr(index%2Bcharacter.length)%3B%0A%7D%3B%0A%0AspinalCase('This%20Is%20Spinal%20Tap')%3B%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

It throw error "potential infiniti loop" but it is not infinite loop. What am I missing?

My code:
```javascript
function spinalCase(str) {
  
  for(var i = 0; i < str.length; i++)
  {
    var char = str.substr(i, 1);
     if(char === char.toUpperCase() && char != " ")
     {
       str = str.replaceAt(i, "-" + char);
     }
  }
  
  str = str.toLowerCase().split(" ").join("-");
  str = str.toLowerCase().split("_").join("-");
  return str;
}

String.prototype.replaceAt=function(index, character) {
    return this.substr(0, index) + character + this.substr(index+character.length);
};

spinalCase('This Is Spinal Tap');

```

**Comment by raisedadead** [comment]
Hi @airsakarya,

If you feel your code doesn't have or will not have any infinite loop case, then simply add `//noprotect` to the top of your code.

This will prevent, the check and you can proceed. Its a safety feature for the code checker, that can be overridden. Nevertheless, be careful that you do not have any infinite loop, or your browser will die! :-D

Any ways this is is being fixed with #5345 , You can read about it in the issue comments!
Hope this helps!

**Comment by ltegman** [comment]
You code does have an infinite loop in it. When the character is uppercase you are replacing it with a dash and the uppercase character, which means the next value of `i` will point to the same uppercase character it just replaced with a dash. This will continue indefinitely, constantly increasing the length of the string and causing an infinite loop.

However, @raisedadead is correct about how to disable loop protection if you think it has happened in error.

In the future please seek help in the [Help Chat](https://gitter.im/FreeCodeCamp/Help) to confirm that what you are seeing is actually a bug and not an issue with your solution.

Thanks and happy coding!
